source/_integrations/zha.markdown

```diff
@@ -432,6 +432,60 @@ Such a custom "ZHA Device Handler" are Python scripts that internally are also r
+      - Try different orientations of the adapter's external antenna (or the whole Zigbee adapter).
+  - USB 3.0 ports/computers/peripherals are known culprits of RFI/EMI/EMF disruption. (See Ref. [1](https://www.usb.org/sites/default/files/327216.pdf) and [2](https://www.unit3compliance.co.uk/2-4ghz-intra-system-or-self-platform-interference-demonstration/)).
+    - Make sure to only connect the Zigbee USB adapter to a USB 2.0 port (and not to a USB 3.x port). 
+      - If your computer only has a USB 3.x port then buy and connect the adapter via a powered USB 2.0 hub:
+        - Using via USB 2.0 hub will in practice convert USB 3.0 to a USB 2.0 port and thus avoid USB 3.0 EMF.
+          - A USB 2.0 hub that uses an external power supply will ensure power requirements are fulfilled.
+    - Shield any unshielded computers/peripherals/devices by adding all-metal enclosures/chassis/casings.
+      - Single-board-computers and USB 3.x hard drives are especially known as sources of EMF/EMI/RFI.
+        - Be aware metal casings can decrease the performance of internal/built-in Zigbee adapters.
+      - Also, be sure to use adequately shielded USB cables for any such peripherals/devices too.
+  - 2.4 GHz RF Interference (RFI) from Wi-Fi Routers and Wi-Fi Access Points or other devices:
+    - First of all, try to place your Zigbee adapter away from Wi-Fi access points or sources of WiFi.
+    - Place Zigbee adapters far away from electrical/power wires/cables, power supplies, and household appliances.
+    - Zigbee could have overlapping frequency ranges with Wi-Fi, see the section above on defining Zigbee channel use.
+  - Add more and decrease the distance between Zigbee devices in Zigbee network mesh to get better range and coverage:
+    - Zigbee uses [mesh networking](https://en.wikipedia.org/wiki/Mesh_networking) and depends on having many "Zigbee Router" devices to extend range and coverage:
+      - Recommendation is to add additional mains-powered Zigbee devices known to be good Zigbee Router devices.
+        - Add more Zigbee Router devices and reduce their distances to extend network mesh coverage and range.
+          - Note that not all mains-powered devices have firmware that makes them act as a Zigbee Router device.
+          - Some brands/models of Zigbee Router devices are known to only work well with the same brand of devices.
+      - Buy a few known good dedicated Zigbee Router products (for example the “[IKEA Tradfri Signal Repeater](https://www.google.com/search?q=IKEA+Tradfri+Signal+Repeater)”).
+        - Search community forums for more "Zigbee signal repeater" or "Zigbee range extender" Zigbee Router tips.
+      - Buy a few additional new Zigbee USB adapters to use after re-flashing them with Zigbee Router firmware.
+        - Reflash/reuse modern Zigbee USB adapters to act as Zigbee Router devices by changing Zigbee firmware.
```

Comment:
This really doesn't work as a big list. For example:

![image](https://user-images.githubusercontent.com/195327/212767828-6e87ff4d-f517-400c-8b4f-fc89db894f86.png)

Not even to start reading this one mobile 🙈  This needs to be structured differently.

Reply:
Got any specific suggestions on how to make it better? That is the same format that got accepted for Bluetooth https://www.home-assistant.io/integrations/bluetooth/#bluetooth-interference-with-other-devices

Reply:
> Not even to start reading this one mobile 🙈 This needs to be structured differently.

Is that really a showstopper? It is not better to now just push this into the ZHA documentation as-is and then improve it?

I think that some documentation about this very important subject of interference avoidance must be better than none.

Reply:
Tried again to restructure that last segment as best I could.